haystack/__init__.py

```diff
@@ -2,57 +2,33 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-import sys
```

Comment:
slightly related: after intense debugging, I found that the current `haystack/__init__.py`, which mixes eager and lazy imports, somehow overshadows the static type of `component`. This means that we would not able to check Protocol adherence of `run` methods decorated with `@component.output_types`.

For this reason, I decided to go back to eager imports in this module.

Reply:
Is this a bug when using all protocols based on components? If so it could be good to bring this change in a separate PR.

Reply:
Yes. I will do this in a different PR.

BTW, this PR is just a POC and not meant to be merged.